portfolio/src/main/java/com/google/sps/servlets/DiscussionServlet.java

```diff
@@ -12,46 +13,73 @@
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.sps.data.Post;
 
 
 @WebServlet("/discussion")
 public class DiscussionServlet extends HttpServlet {
 
+  private ArrayList<String> postTitles;
+
+  @Override
+  public void init() {
+    postTitles = new ArrayList<>();
```

Comment:
You shouldn't need to use both a class variable to store data, and the datastore. All of the data should exist in the datastore, so see if you can do this without the need for this class variable

Reply:
hmm..okay the problem i got with that was that even though all of the posts are stored on the datastore, I don't know how to retrieve them without knowing exactly what the user said. The idea behind was atleast storing the post titles so I know how to retrieve them.

Reply:
Right so I think instead of storing things under the title, you should store everything in a table named "post" or similar, and include both the title and the content in that entity.  

The issue with doing this this way, is that the postTitles variable would only last until the server is reset.  The database will persist permanently, but if you were to deploy this app in production, but then have to reset it (for example to update it), then the postTitles arrayList would be cleared out.

lmk if you want to hop on a chat to discuss this